src/DocRaptor.php

```diff
@@ -39,15 +40,17 @@ class ApiWrapper
     protected $baseurl;
 
 
-
     /**
+     * @param HttpTransferInterface $httpClient
      * @param string|null $api_key
      */
-    public function __construct($api_key = null)
+    public function __construct(HttpTransferInterface $httpClient, $api_key = null)
```

Comment:
Yeah, based on Example #4 in [this documentation](http://php.net/manual/en/functions.arguments.php), I think you are correct regarding calling `new SomeClass()` in a function def.

I am pro lazy instantiation. Also, I would like to reverse the order of params so that one can pass an api key and not an http client, because I think that is the common case.